diff.py

```diff
@@ -1575,6 +1575,13 @@ def __init__(self, config: Config) -> None:
         super().__init__(config)
         self.seen_jr_ra = False
 
+    def _normalize_arch_specific(self, mnemonic: str, row: str) -> str:
+        if mnemonic == "li":
+            return re.sub(
+                f"(-?0x[0-9a-fA-F]+)", lambda m: str(int(m.group(1), 16)), row
+            )
```

Comment:
Using a compiled regex with `re.compile` (https://docs.python.org/3/library/re.html#re.compile) would be better